integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/Hooks.java

```diff
@@ -39,42 +38,45 @@ public void before(Scenario scenario) {
-                                    .until(() -> ProcessorResource.getProcessorList(token, bridgeId)
-                                            .getSize() == 0);
-                        }
-                    }
-                    switch (bridge.getStatus()) {
-                        case ACCEPTED:
-                        case PROVISIONING:
-                        case READY:
-                        case FAILED:
-                            try {
-                                BridgeResource.deleteBridge(token, bridgeId);
-                            } catch (Exception e) {
-                                LOGGER.warn(e, () -> "Unable to delete bridge with id " + bridgeId);
+        if (!Boolean.parseBoolean(DISABLE_CLEANUP)) {
+            String token = Optional.ofNullable(context.getManagerToken()).orElse(BridgeUtils.retrieveBridgeToken());
+            // Remove all bridges/processors created
+            context.getAllBridges().values()
+                    .stream()
+                    .filter(bridgeContext -> !bridgeContext.isDeleted())
+                    .forEach(bridgeContext -> {
+                        final String bridgeId = bridgeContext.getId();
+                        BridgeResponse bridge = BridgeResource.getBridgeDetails(token, bridgeId);
+                        if (bridge.getStatus() == ManagedResourceStatus.READY) {
+                            ProcessorListResponse processorList = ProcessorResource.getProcessorList(
+                                    token,
+                                    bridgeId);
+                            if (processorList.getSize() > 0) {
+                                processorList.getItems().stream().forEach(
+                                        p -> ProcessorResource.deleteProcessor(token, bridgeId,
+                                                p.getId()));
+                                Awaitility.await()
+                                        .atMost(Duration.ofMinutes(2))
+                                        .pollInterval(Duration.ofSeconds(5))
+                                        .until(() -> ProcessorResource.getProcessorList(token, bridgeId)
+                                                .getSize() == 0);
                             }
-                        default:
-                            break;
-                    }
-                });
+                        }
+                        switch (bridge.getStatus()) {
+                            case ACCEPTED:
+                            case PROVISIONING:
+                            case READY:
+                            case FAILED:
+                                try {
+                                    BridgeResource.deleteBridge(token, bridgeId);
+                                } catch (Exception e) {
+                                    context.getScenario().log("Unable to delete bridge with id " + bridgeId
```

Comment:
Please propagate the exception. It would be useful to know that something is not correct with removing of bridges.

Reply:
Not sure it should be relevant to fail the test if the cleanup has failed somehow...
@sutaakar wdyt ?

Reply:
Well, it is unrelated to the test itself. On the otherwise we should be informed of such issues as they indicate that something is broken. If we just log it then we easily overlook such issues.
I think that we cloud throw the error to indicate that some issue happened, even when not directly related to the test.

Reply:
updated